### ISSUE TITLE
does it utilize twitter's url shortening?

### ISSUE THREAD

**Issue by corb** [jpurcell/birdhouse#3]
This addon works great!

I was wondering if this addon utilized twitter's url shortening?

(ie, "data-url" found in the share button (from http://dev.twitter.com/pages/tweet_button))

I've been poking around the code/example/faqs, but couldn't see anything, so I figured I'd come straight to the source

**Comment by jpurcell** [comment]
Great question! I just commented and was completely wrong, so let me try again. The implementation I'm currently using on an app shortens the link before opening the tweet() box, hence my incorrect hasty reply!

BirdHouse currently does not shorten links. That would be a great feature though, perhaps a "short_tweet" function that auto-shortens the link, displays the shortened tweet, and then prompts for posting. Would this be a feature you're interested in?

**Comment by corb** [comment]
Sorry for the follow up...does this mean the birdhouse addon can use the twe.ly? If so, can you point me to where I can find examples/source/etc?

Thanks again, this addon is such a life saver

**Comment by jpurcell** [comment]
Hahaha, looks like you replied before I could update my comment!

It currently does not use a URL shortener. I could easily incorporate that--see my updated comment.

**Comment by corb** [comment]
I would _love_ that feature! Very interested

**Comment by jpurcell** [comment]
Alright, I'll put that in the works for v0.8! Hopefully to be released within the next few days. Great suggestion!

**Comment by johnrgoslin** [comment]
Please feel free to use my library I created called tinify.js

**Comment by jpurcell** [comment]
Hey jgoslin, great library! I already incorporated twe.ly, but if anyone else is interested that repo is at http://github.com/jgoslin/tinify.js. To incorporate it with BirdHouse, just place the tinify function in the shorten_url function.
